dnf/repo.py

```diff
@@ -886,7 +886,9 @@ def load(self):
 
         """
         if self.metadata or self._try_cache():
-            if self._check_config_file_age and self.repofile \
+            if self.metadata_expire != -1 \
```

Comment:
I think the original logic was correct and your suggestion is wrong. The metadata can be expired if metadata_expire != -1. If your suggestion was >= 0, it would make more sense, but anyway, this is not in line with the documentation [[1]].


[1]: https://github.com/rpm-software-management/dnf/blame/master/doc/conf_ref.rst#L425